client/flutter/lib/home_page.dart

```diff
@@ -33,8 +36,11 @@ class HomePage extends StatelessWidget {
                 PageButton(
                   Color(0xff3b8bc4),
                   S.of(context).homePagePageButtonProtectYourself,
-                  () => Navigator.of(context).push(
-                      MaterialPageRoute(builder: (c) => ProtectYourself())),
+                  () {
+                    analytics.logEvent(name: 'Protect_Yourself');
```

Comment:
Why do we need this line if we've already added the navigatorObserver?

Reply:
I think if we use the `navigatorObserver` we also don't need to include the analytics object in the state. Example here: https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_analytics/firebase_analytics/example/lib/main.dart

Reply:
> Why do we need this line if we've already added the navigatorObserver?
NavigatorObserver tries to automatically extract the name of the route('Protect_Yourself') from RouteSettings. But currently we don't have RouteSettings, I expect after v1 we can work on refactoring Navigation & routing. For now, either we have to manually log an event or manually add routesettings like this.
```
Navigator.of(context).push(MaterialPageRoute(
                settings: const RouteSettings(name: 'Protect_Yourself'),
                builder: (BuildContext context) => ProtectYourself()
                );
```
It's more clear to explicitly log events for now.

Reply:
> I think if we use the `navigatorObserver` we also don't need to include the analytics object in the state. Example here: https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_analytics/firebase_analytics/example/lib/main.dart

I'm not clear what you mean, but navigationObserver will only log navigation changes. With the analytics object we can log everything we need. Regarding 'state' I just passed down the object, I'm expecting a state management approach after v1. So we can refactor then.

Reply:
Makes sense!

Reply:
> > Why do we need this line if we've already added the navigatorObserver?
> > NavigatorObserver tries to automatically extract the name of the route('Protect_Yourself') from RouteSettings. But currently we don't have RouteSettings, I expect after v1 we can work on refactoring Navigation & routing. For now, either we have to manually log an event or manually add routesettings like this.
> 
> ```
> Navigator.of(context).push(MaterialPageRoute(
>                 settings: const RouteSettings(name: 'Protect_Yourself'),
>                 builder: (BuildContext context) => ProtectYourself()
>                 );
> ```
> 
> It's more clear to explicitly log events for now.

@ShivamSinghania sounds good. Can you add a similar tracking code for all the page transitions?